### PR TITLE
Add Staryu macropad support

### PR DESCRIPTION
**Pull request by krusli**
Ported [Staryu (5 key) macropad](https://www.aliexpress.com/item/Staryu-Mini-Macro-Pad-custom-keyboard-by-tkg-kai-Mini-macro-pad-mechanical-keyboard-5-keys/32817560746.html) code from https://github.com/kairyu/tmk_keyboard_custom.

~~READMEs and backlighting support in progress.~~

Backlighting support is rudimentary (no dimming/PWM) as the Staryu has individually addressable backlight LEDs, and QMK does not seem to support having several pins for backlighting. I took the approach similar to the default keymap for the M6-A: simply have functions (in `backlight_staryu.h` instead of the keymap in this case) that the user can use to individually turn the LEDs on and off.

### CODE REVIEW

File: keyboards/staryu/staryu.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef KB_H
+#define KB_H
+
+#include "quantum.h"
+
+#define KEYMAP( \
```
Comment (drashna) [inline keyboards/staryu/staryu.h]: We're deprecating KEYMAP in favor of LAYOUT. Change all KEYMAP to LAYOUT?

File: keyboards/staryu/keymaps/krusli/keymap.c [hunk 2]
```diff
@@ -14,7 +14,7 @@ GNU General Public License for more details.
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
-#include QMK_KEYBOARD_H
+#include "staryu.h"
```
Comment (drashna) [inline keyboards/staryu/keymaps/krusli/keymap.c]: Actually, `#include QMK_KEYBOARD_H` is prefered, as it's keyboard agnostic.

File: keyboards/staryu/staryu.h (hunk 1 above)
Comment (krusli) [reply keyboards/staryu/staryu.h]: Done @ 909340d649db021d319294627e7a00be6e646947

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**